### PR TITLE
option to install prerequisites

### PR DESCRIPTION
**Pull request by rijobro**
fixes https://github.com/SyneRBI/SyneRBI_VM/issues/156

@johannesmayer did you say you'd be able to try this one?

### CODE REVIEW

File: scripts/UPDATE.sh [hunk 1]
```diff
@@ -142,6 +145,17 @@ git config --global filter.nbstripout.extrakeys '
   metadata.celltoolbar metadata.language_info.codemirror_mode.version
   metadata.language_info.pygments_lexer metadata.language_info.version'
 
+# Optionally install pre-requisites
+if [ $apt_install == 1 ]; then
+  cd ~/devel/SyneRBI_VM
+  git pull
```
Comment (KrisThielemans) [inline scripts/UPDATE.sh:151]: delete this. Already done when this scripts get called.

File: scripts/UPDATE.sh [hunk 2]
```diff
@@ -35,19 +35,22 @@ trap 'echo An error occurred in $0 at line $LINENO. Current working-dir: $PWD' E
...
   t) SB_TAG=$OPTARG;;
   j) num_parallel=$OPTARG;;
   h)
-   echo "Usage: $0 [-t tag] [-j n] [-r] "
+   echo "Usage: $0 [-t tag] [-j n] [-r] [-s]"
    echo "Use the tag option to checkout a specific version of the SIRF-SuperBuild."
    echo "   Otherwise the most recent release will be used."
    echo "Use the -j option to change the number of parallel builds from the default ${num_parallel}"
    echo "Use the -r option to reset your git remotes to default SyneRBI sources."
+   echo "Use the -s option to run apt install for all prerequisites (use with caution)."
```
Comment (KrisThielemans) [inline scripts/UPDATE.sh:53]: can you find a way to generalise this beyond `apt` in a few words? (as it's doing more)

File: documentation/Updating-VM-to-a-new-major-release.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # Updating the VM
-In most cases, typing `update_VM.sh` in a terminal will be sufficient to update the software. However, for some upgrades, extra system or Python packages can be required which were not preinstalled on the VM of a previous version. You can install these as follows
+In most cases, typing `update_VM.sh` in a terminal will be sufficient to update the software. However, for some upgrades, extra system or Python packages can be required which were not preinstalled on the VM of a previous version. You can install these by adding the `-s` argument to the update command: `update_VM.sh -s`. In essence, this will run the following:
```
Comment (KrisThielemans) [inline documentation/Updating-VM-to-a-new-major-release.md:2]: let's add that they might need a `sudo apt update` before that.

**Comment by johannesmayer** [comment]
> fixes #156
> 
> @johannesmayer did you say you'd be able to try this one?

I checked out this branch, and performed the steps listed in the documentation on the petri-dish example of a v2.1 VM in which  I had previously ran the update_VM.sh.

Now I ran the update_VM.sh -s and it had an error saying:

```
...

fatal: repository 'https://itk.org/ITK.git/' not found
-- Had to git clone more than once:
          3 times.
CMake Error at /home/sirfuser/devel/buildVM/builds/ITK/tmp/ITK-gitclone.cmake:66 (message):
  Failed to clone repository: 'https://itk.org/ITK.git'


CMakeFiles/ITK.dir/build.make:89: recipe for target 'builds/ITK/stamp/ITK-download' failed
make[2]: *** [builds/ITK/stamp/ITK-download] Error 1
CMakeFiles/Makefile2:407: recipe for target 'CMakeFiles/ITK.dir/all' failed
make[1]: *** [CMakeFiles/ITK.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
-- Boost version: 1.65.1
-- Found the following Boost libraries:
....
```

**Comment by KrisThielemans** [comment]
interesting. @paskino mentioned something similar. The strange thing is that we've switched ITK_URL to https://github.com/InsightSoftwareConsortium/ITK ~5months ago. are you sure you ran
```
update_VM -s -t origin/master
```

**Comment by rijobro** [comment]
could it be that the URL didn't update if it's cached?

**Comment by KrisThielemans** [comment]
don't think so.
https://github.com/SyneRBI/SyneRBI_VM/blob/80e4ce94e1cad64e72b2cc7f751386321d9ab719/scripts/UPDATE.sh#L174-L176

I suspect that @johannesmayer ran it without the tag, in which case it will resort back to 2.1, which had the out-dated URL presumably,

**Comment by johannesmayer** [comment]
> don't think so.
> https://github.com/SyneRBI/SyneRBI_VM/blob/80e4ce94e1cad64e72b2cc7f751386321d9ab719/scripts/UPDATE.sh#L174-L176
> 
> I suspect that @johannesmayer ran it without the tag, in which case it will resort back to 2.1, which had the out-dated URL presumably,

Yes that's entirely correct, I didn't run it with the tag. Now I did and it worked and performed the update script without any error.

**Comment by KrisThielemans** [comment]
Excellent. thanks for testing!

@rijobro could you address the doc-related comments? Also, please raise an issue that we can now probably simplify the `bootstrap_ubunty.sh` script and/or vagrant stuff. But doing that will be for  later.

**Comment by KrisThielemans** [comment]
maybe hang-on until merging till we understand #160

**Pull request merged**